src/utils/storage/useStoredState.ts

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react'
+import local from './local'
+
+const useStoredState = <T>(key: string): [T | undefined, React.Dispatch<React.SetStateAction<T>>] => {
```

Comment:
Maybe this should be name something like `useLocalState` just to be clear that it isn't involving the `sessionStorage`.

Reply:
"Local" state, to me, doesn't imply that it's stored in a permanent place. I would keep this name.